bitnami/apisix/README.md

```diff
@@ -102,7 +104,7 @@ dataPlane:
         name: apisix-routes
   extraVolumeMounts:
     - name: routes
-      mountPath: /opt/bitnami/apisix/conf/apisix.yaml
+      mountPath: /usr/local/apisix/conf/apisix.yaml
```

Comment:
This is not necessary, there is a symlink in the docker image that links these 2 folders

https://github.com/bitnami/containers/blob/main/bitnami/apisix/3/debian-12/Dockerfile#L48

Reply:
Thank you for the feedback!

I changed this because I was hitting an error with the default path. Error is below:
```
...
2024/06/11 13:48:01 [error] 10#10: *5651 [lua] config_yaml.lua:73: failed to fetch /usr/local/apisix/conf/apisix.yaml attributes: cannot obtain information from file '/usr/local/apisix/conf/apisix.yaml': No such file or directory, context: ngx.timer
2024/06/11 13:48:01 [error] 13#13: *5655 [lua] config_yaml.lua:73: failed to fetch /usr/local/apisix/conf/apisix.yaml attributes: cannot obtain information from file '/usr/local/apisix/conf/apisix.yaml': No such file or directory, context: ngx.timer
2024/06/11 13:48:01 [error] 7#7: *5659 [lua] config_yaml.lua:73: failed to fetch /usr/local/apisix/conf/apisix.yaml attributes: cannot obtain information from file '/usr/local/apisix/conf/apisix.yaml': No such file or directory, context: ngx.timer
2024/06/11 13:48:01 [warn] 7#7: *5321 [lua] config_yaml.lua:127: wait for more time, context: ngx.timer
...
```

When the mountPath is set to `/opt/bitnami/apisix/conf/apisix.yaml`, I can see the path there when exec-ing into the container

```
apisix@apisix-data-plane-cb89748d8-l72nz:/$ ls -la /opt/bitnami/apisix/conf/
total 76
drwxrwxr-x 1 root root  4096 Jun 11 13:45 .
drwxr-xr-x 1 root root  4096 Jun  6 10:59 ..
-rw-r--r-- 1 root 1001   118 Jun 11 13:45 apisix.yaml
drwxr-xr-x 2 root root  4096 May 13 15:17 cert
-rwxr-xr-x 1 root root 39191 May 13 14:48 config-default.yaml
-rwxr-xr-x 1 root root  2170 May 13 14:48 config.yaml
-rwxr-xr-x 1 root root  1535 May 13 14:48 debug.yaml
-rwxr-xr-x 1 root root  5290 May 13 14:48 mime.types
``` 

But not under /usr/local/apisix

```
ls -la /usr/local/apisix/conf/
total 88
drwxr-sr-x 3 apisix 1001  4096 Jun 11 13:45 .
drwxrwsrwx 9 root   1001  4096 Jun 11 13:45 ..
-rw-r--r-- 1 apisix 1001    36 Jun 11 13:45 apisix.uid
drwxr-sr-x 2 apisix 1001  4096 Jun 11 13:45 cert
-rwxr-xr-x 1 apisix 1001 39191 Jun 11 13:45 config-default.yaml
-rw-r--r-- 1 apisix 1001  1218 Jun 11 13:45 config.yaml
-rwxr-xr-x 1 apisix 1001  1535 Jun 11 13:45 debug.yaml
-rwxr-xr-x 1 apisix 1001  5290 Jun 11 13:45 mime.types
-rw-r--r-- 1 apisix 1001 12429 Jun 11 13:45 nginx.conf
```

Looking under /usr/local and /opt/bitnami, it doesn't look like those paths are symlinked. Do you think there may be an issue with the container?

Reply:
Oh, I think I see what is breaking the symlink. The contents of `/usr/local/apisix` are overwritten by a volumeMount at https://github.com/bitnami/bitnami-charts/blob/b9e533cb5bc8ee31882457136c00e4d253941797/bitnami/apisix/templates/data-plane/dep-ds.yaml#L176.

Reply:
This line is copying the folder again instead of creating the symlink. 

https://github.com/bitnami/charts/blob/main/bitnami/apisix/templates/_helpers.tpl#L335

That logic was generated before updating the security of the charts and adding the emptyDir volumes during the initialization. Would you mind changing it and adding it to this PR? Please remember to undo the /usr/local changes in this file after updating the _helpers file.

Reply:
Taking a closer look, I think this fix may be slightly more involved than swapping that line out for a symlink. I attempted to swap the copy with the symlink like so:

```
diff --git a/bitnami/apisix/templates/_helpers.tpl b/bitnami/apisix/templates/_helpers.tpl
index 5ec11eb174..1caa079bfb 100644
--- a/bitnami/apisix/templates/_helpers.tpl
+++ b/bitnami/apisix/templates/_helpers.tpl
@@ -332,7 +332,7 @@ Init container definition for waiting for the database to be ready
     - -ec
     - |
       #!/bin/bash
-      cp -R /opt/bitnami/apisix/conf /usr/local/apisix
+      ln -sf /opt/bitnami/apisix/conf /usr/local/apisix
       ln -sf /opt/bitnami/apisix/deps /usr/local/apisix
       ln -sf /opt/bitnami/apisix/openresty/luajit/share/lua/*/apisix /usr/local/apisix
       mkdir -p /usr/local/apisix/logs
```

Since /usr/local/apisix/conf is a symlink to /opt/bitnami/apisix/conf, the pre-render-config.yaml and config.yaml created at https://github.com/bitnami/charts/blob/main/bitnami/apisix/templates/_helpers.tpl#L340 and https://github.com/bitnami/charts/blob/main/bitnami/apisix/templates/_helpers.tpl#L341 would be stored under /opt/bitnami/conf, which isn't in a volume (and I'm also currently getting the error "bash: line 7: /usr/local/apisix/conf/pre-render-config.yaml: Read-only file system", but I think that's solvable).

I think we would have to add another volume and volume mount for /opt/bitnami if we want to symlink /usr/local/apisix/conf to /opt/bitnami/apisix/conf and store the generated configs under /usr/local/apisix/conf, unless I'm missing something.

Reply:
ok, using the symlink as we have in the container will require changes in the code. Let's keep your changes as there are many other references to /usr/local/apisix in the templates and we can review this symlink configuration in the future. Thanks for reviewing the configuration.